### PR TITLE
[VL] fallback text file format hive scan

### PR DESCRIPTION
**Pull request by yma11**
## What changes were proposed in this pull request?

fallback text file format hive scan as it's not supported in Velox

## How was this patch tested?

N/A

### CODE REVIEW

**Comment by yma11** [comment]
@zhouyuan can you help review this PR? this is a bug fix as we should not offload text file.

**Comment by zhouyuan** [comment]
@yma11 the code looks good for velox, but it's not working on ClickHouse backend, can you make it work with Velox backend only?

```
14:22:55  - GLUTEN-2180: Test data field type not match *** FAILED ***
14:22:55    java.lang.AssertionError: assertion failed: FallBack true check error: Scan hive default.test_table_2180 [a#324133, b#324134, c#324135, d#324136, e#324137], HiveTableRelation [`default`.`test_table_2180`, org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe, Data Cols: [a#324133, b#324134, c#324135, d#324136, e#324137], Partition Cols: []]
14:22:55    at scala.Predef$.assert(Predef.scala:223)
14:22:55    at io.glutenproject.execution.WholeStageTransformerSuite$.checkFallBack(WholeStageTransformerSuite.scala:305)
14:22:55    at io.glutenproject.execution.WholeStageTransformerSuite.compareResultsAgainstVanillaSpark(WholeStageTransformerSuite.scala:255)
14:22:55    at io.glutenproject.execution.GlutenClickHouseHiveTableSuite.$anonfun$new$51(GlutenClickHouseHiveTableSuite.scala:801)
```

File: gluten-core/src/main/scala/org/apache/spark/sql/hive/HiveTableScanExecTransformer.scala [hunk 1]
```diff
@@ -132,26 +132,10 @@ class HiveTableScanExecTransformer(
...
-
     fileFormat match {
       case ReadFileFormat.JsonReadFormat => ValidationResult.ok
       case ReadFileFormat.TextReadFormat =>
-        if (!hasComplexType) {
-          ValidationResult.ok
-        } else {
-          ValidationResult.notOk("does not support complex type")
-        }
+        ValidationResult.notOk("Text file format is not supported")
```
Comment (zhouyuan) [inline gluten-core/src/main/scala/org/apache/spark/sql/hive/HiveTableScanExecTransformer.scala:138]: maybe do a check on the backend first?

**Comment by yma11** [comment]
Doesn't need anymore.

**Pull request closed without merging**